buildpack.go

```diff
@@ -412,16 +438,18 @@ type DependencyResolver struct {
 
 	// StackID is the stack id of the build.
 	StackID string
+
+	logger bard.Logger
 }
 
 // NewDependencyResolver creates a new instance from the buildpack metadata and stack id.
-func NewDependencyResolver(context libcnb.BuildContext) (DependencyResolver, error) {
+func NewDependencyResolver(context libcnb.BuildContext, logger bard.Logger) (DependencyResolver, error) {
```

Comment:
cc: @dmikusa 
Wouldn't this break all calls to NewDependencyResolver?

Could we instead leave the logger added to the DependencyResolver struct (changing to a public property with logger -> Logger) and have users initialize it when necessary via:

```
dr, err := libpak.NewDependencyResolver(context)
...
dr.Logger = <b.Logger>
```

Reply:
Yes, it would need some adjustments. I've seen that in the code multiple times, but I have to admit that I don't like it very much. The constructor should bring an objects into a usable state. 

Alternatively we could extract the functionality and do something like that (on the caller side):

```go
checker := libpak.NewDependencyChecker(&b.Logger)

depJDK, err := dr.Resolve("jdk", v)
if (jdkRequired && !nativeImage) && err != nil {
   return libcnb.BuildResult{}, fmt.Errorf("unable to find dependency\n%w", err)
}

checker.Check(depJDK)
```

Reply:
I agree it's best if the constructor makes a usable object, but we can't break the API. There are also several place where the Logger is set in this manner. To keep the API stable, I'd also be fine doing that here. It's definitely on my todo list so this is changed in v2.

Other options:

1. Make a totally new `NewDependencyResolverWithLogger`.
2. Use the [option pattern](https://github.com/paketo-community/cargo/blob/9a30f3c5983f7e3b03bde63f1cae4b5d88c376cc/cargo/cargo.go#L36). `NewDependencyResolver(context libcnb.BuildContext, options ...Option)`.  This way the options are not required so the API is consistent, but one can pass through things like the logger. It doesn't require one to pass through anything that optional though, which may not really solve the problem here. If a Logger is required, then a user could still forget to include it and cause problems.

Reply:
That makes sense and as it's already on your list for `v2`, I'm fine with using the easiest route.

Reply:
And FYI, I think we'll probably shoot for a v2 after libcnb releases it's v2. That way libpak v2 matches with libcnb v2.